File: tailoring-online-services/store-management-service/src/main/java/com/store_management_service/dto/StoreDto.java
package com.store_management_service.dto;

import com.store_management_service.model.Tailor;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class StoreDto implements Serializable {
    private String name;
    private String description;
    private Map<String, String> type;
    private List<String> images;
    private String logo;
    private String coverImage;
    private Double rating;
    private Tailor tailor;
}

File: tailoring-online-interfaces/src/app/core/dtos/store.dto.ts
import { Tailor } from "../interfaces/tailor.interface";

export interface StoreDto {
  name: string;
  description: string;
  type?: { [key: string]: string };
  images?: string[];
  logo: string;
  coverImage: string;
  rating: number;
  tailor: Tailor;
}
